### PR TITLE
multi-gpu support for prefix-tuning

### PR DESCRIPTION
**Pull request by alexanderhanboli**
The current way of initializing `input_tokens` in `__init__` and using `device = next(self.parameters()).device` to find device does not work for multi-gpu.

Instead we should use this [fix](https://github.com/huggingface/transformers/blob/1c220ced8ecc5f12bc979239aa648747411f9fc4/src/transformers/modeling_utils.py#L120) to find the device of the module and assign it to the input tokens.

### CODE REVIEW

File: src/transformers/adapters/prefix_tuning.py [hunk 1]
```diff
@@ -80,12 +80,11 @@ def __init__(
         self.dropout = nn.Dropout(self.config.dropout)
 
     def forward(self, batch_size):
-        device = next(self.parameters()).device
         key_values = (
             self.control_trans.unsqueeze(0)
             .expand(batch_size, -1)
             .view(batch_size, self.config.prefix_length, self.n_layers * 2, self.n_heads, self.n_embd_per_head)
-            .to(device)
+            .to(self.device)
```
Comment (calpt) [inline src/transformers/adapters/prefix_tuning.py:87]: Please also add the mixin to the `FlatPrefixTuning` class to pass the tests.

**Review by calpt — COMMENTED** [review]
Hey @alexanderhanboli, thanks for looking into this issue and providing a fix! Happy to merge this as soon as all the tests are passing.

(I'm assuming this closes #358)

File: src/transformers/adapters/prefix_tuning.py (hunk 1 above)
Comment (alexanderhanboli) [reply src/transformers/adapters/prefix_tuning.py:87]: Hey @calpt sorry for the delay. I just added the mixin to that class.

**Review by calpt — APPROVED** [review]
Looks good, thanks!

**Pull request merged**